tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/config/TkmsConfiguration.java

```diff
@@ -145,10 +148,20 @@ public TkmsMessageSerializer tkmsMessageSerializer() {
 
   @Bean
   @ConditionalOnMissingBean(IEnvironmentValidator.class)
-  public EnvironmentValidator tkmsMigrationHandler() {
+  @ConditionalOnClass(name = "jakarta.validation.Validator")
+  @ConditionalOnBean(type = "jakarta.validation.Validator")
+  public EnvironmentValidator tkmsEnvironmentValidator() {
     return new EnvironmentValidator();
   }
 
+  @Bean
+  @ConditionalOnMissingBean(IEnvironmentValidator.class)
+  @ConditionalOnClass(name = "javax.validation.Validator")
+  @ConditionalOnBean(type = "javax.validation.Validator")
```

Comment:
I actually like that approach.
I wonder though if we are going to do it more often, than in one lib, should this go to some kind of `base-utils` class, and expose `IValidatable` interface for libs and service to implement?

Then this autoconfiguration can be done in a single place.

Reply:
Maybe, we don't have too many of those tbh, probably 5 total. And it is supposed to go away.